src/Components/common/FacebookLoginButton/FacebookLoginButton.js

```diff
@@ -0,0 +1,63 @@
+import React, { useContext } from "react";
+import FacebookLogin from "react-facebook-login/dist/facebook-login-render-props";
+import { store as loginStore } from "../../Login/LoginContainer";
+import { store as registerStore } from "../../Register/RegisterContainer";
+import { ReactComponent as FacebookLogoIcon } from "../../../assets/facebook-logo.svg";
+import { HANDLE_SOCIAL_LOGIN } from "../../../utils/action-types";
+
+export const FacebookLoginButton = () => {
+  const facebookLoginEnabled =
+    window.Pelcro.site.read()?.facebook_app_id;
+
+  const { dispatch: loginDispatch } = useContext(loginStore);
+  const { dispatch: registerDispatch } = useContext(registerStore);
+
+  const onSuccess = (facebookUser) => {
+    loginDispatch?.({
```

Comment:
why are we always dispatching events in both the login and the register containers? I think I'm misunderstanding something

Reply:
Instead of creating RegisterFacebookLoginButton & LoginFacebookLoginButton 
I'm using the same component under both containers, so it dispatches both. and one of them arrives.